src/main/resources/project/server/deployApplication.pl

```diff
@@ -0,0 +1,446 @@
+# preamble.pl
+$[/myProject/procedure_helpers/preamble]
+
+my $PROJECT_NAME = '$[/myProject/projectName]';
+my $PLUGIN_NAME = '@PLUGIN_NAME@';
+my $PLUGIN_KEY = '@PLUGIN_KEY@';
+use Data::Dumper;
+use URI;
+use File::Basename;
+use JSON;
+
+$| = 1;
+
+main();
+
+sub main {
+    my $jboss = EC::JBoss->new(
+        project_name                    => $PROJECT_NAME,
+        plugin_name                     => $PLUGIN_NAME,
+        plugin_key                      => $PLUGIN_KEY,
+        no_cli_path_in_procedure_params => 1
+    );
+
+    my $params = $jboss->get_params_as_hashref(qw/
+        applicationContentSourcePath
+        deploymentName
+        runtimeName
+        enabledServerGroups
+        disabledServerGroups
+        additionalOptions
+        /);
+
+    my $param_application_content_source_path = $params->{applicationContentSourcePath};
```

Comment:
Why do we need additional variables if we already have $params hashref?

Reply:
It just for easier usage within IDE (exact auto completion by param perfix, without possible auto completion by keys). PS. Sure once we move to usage of many classes and their objects, we may forget about auto completion, but it is another story.